### PR TITLE
Rename `Users`-Collection to `Actors` to be more specific

### PR DESCRIPTION
**Pull request by pfefferle**
/cc @Menrath

### CODE REVIEW

**Review by mattwiebe — DISMISSED** [review]
"Oh no this is going to cause so many problems on wpcom, we use the `Users` collection in a few places"

"Why isn't the users class deleted?"

"Oh this is nice back compat work" 😅

**Comment by Menrath** [comment]
Very nice! :) A consistent and specific language in ActivityPub is so important to improve collaboration and maintenance.

**Comment by pfefferle** [comment]
> Idea: What about adding deprecation notices to the Users collection right away?

@Menrath what do you mean by that? I added deprecation messages to all functions in the Users Collection, is there an easier way?

**Comment by Menrath** [comment]
Never mind! I didn't see the _deprecated... function, I was just looking for `https://developer.wordpress.org/reference/functions/_doing_it_wrong/` ;)

File: includes/collection/class-actors.php [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+		switch ( $user_id ) {
+			case self::BLOG_USER_ID:
+				return new Blog();
+			case self::APPLICATION_USER_ID:
+				return new Application();
+			default:
+				return User::from_wp_user( $user_id );
+		}
+
+		return new WP_Error(
```
Comment (obenland) [inline includes/collection/class-actors.php]: I don't know if this statement ever reached? The `default` in the `switch` statement should take precedent?

File: includes/collection/class-actors.php [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+			'activitypub_user_not_found',
+			\__( 'Actor not found', 'activitypub' ),
+			array( 'status' => 404 )
+		);
+	}
+
+	/**
+	 * Get the Actor by username.
+	 *
+	 * @param string $username The Actor-Name.
```
Comment (obenland) [inline includes/collection/class-actors.php]:
```suggestion
	 * @param string $username Name of the Actor.
```
Considered `Actorname` but that is probably not a thing.

File: includes/collection/class-actors.php [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+			'activitypub_user_not_found',
+			\__( 'Actor not found', 'activitypub' ),
+			array( 'status' => 404 )
+		);
+	}
+
+	/**
+	 * Get the Actor by resource.
+	 *
+	 * @param string $uri The Actor-Resource.
```
Comment (obenland) [inline includes/collection/class-actors.php]:
```suggestion
	 * @param string $uri The Actor resource.
```

File: includes/collection/class-actors.php (hunk 2 above)
Comment (obenland) [reply includes/collection/class-actors.php]:
```suggestion
	 * @param string $username The Actor name.
```
Or this?

File: includes/collection/class-actors.php [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+					\__( 'Wrong scheme', 'activitypub' ),
+					array( 'status' => 404 )
+				);
+		}
+	}
+
+	/**
+	 * Get the Actor by resource.
+	 *
+	 * @param string $id The Actor-Resource.
```
Comment (obenland) [inline includes/collection/class-actors.php]:
```suggestion
	 * @param string $id The Actor resource.
```

File: includes/collection/class-actors.php (hunk 1 above)
Comment (pfefferle) [reply includes/collection/class-actors.php]: That was a PR and I really missed that! thanks 👍

**Pull request merged**